subsys/bluetooth/mesh/friend.c

```diff
@@ -67,6 +67,8 @@ static struct bt_mesh_adv *adv_alloc(int id)
 	return &adv_pool[id].adv;
 }
 
+static sys_slist_t friend_cb_list = SYS_SLIST_STATIC_INIT(&friend_cb_list);
```

Comment:
I think it's worth considering whether we should handle this through a dedicated linker section instead, since then the whole stuffy can go into ROM instead of RAM. That said, we should consider this holistically for Mesh (and even other Bluetooth) APIs. One thing you do loose then is the possibility to have an `unregister()` function, but I see you didn't even try to add it here, so probably not an issue.